### PR TITLE
[DEVTOOLING-267] Add PCKE Support to Javascript Platform SDK

### PR DESCRIPTION
**Pull request by themooks**
Adding Support for PKCE Grant (methods to authenticate and to generate PKCE Code Verifier and to compute Code Challenge)

### CODE REVIEW

File: resources/sdk/purecloudjavascript/templates/README.mustache [hunk 1]
```diff
@@ -160,6 +160,71 @@ client.refreshCodeAuthorizationGrant(clientId,clientSecret,refreshToken)
...
+  })
+  .catch((err) => {
+    // Handle failure response
+    console.log(err);
+  });
+```
+
+The SDK provides methods to generate a PKCE Code Verifier and to compute PKCE Code Challenge.
+
+```javascript
```
Comment (dginty4) [inline resources/sdk/purecloudjavascript/templates/README.mustache:179]: Maybe put this code block in the code block above (line 165) just so people know it's related.

File: resources/sdk/purecloudjavascript/templates/README.mustache [hunk 2]
```diff
@@ -160,6 +160,71 @@ client.refreshCodeAuthorizationGrant(clientId,clientSecret,refreshToken)
...
```
Comment (dginty4) [inline resources/sdk/purecloudjavascript/templates/README.mustache:223]: same for here, probably best to put this in the code block above to keep everything together

File: resources/sdk/purecloudjavascript/templates/ApiClient.mustache [hunk 3]
```diff
@@ -417,6 +417,339 @@ class ApiClient {
...
+			return randomString;
+		}
+	}
+
+	/**
+	 * @description Compute Base64Url PKCE Code Challenge from Code Verifier.
+	 * @param {string} code - code verifier used to generate the code challenge
+	 */
+	computePKCECodeChallenge(code) {
+		if (code.length < 43 || code.length > 128) {
```
Comment (dginty4) [inline resources/sdk/purecloudjavascript/templates/ApiClient.mustache:541]: You might need to check that code is defined, I was able to get this:
```
if (code.length < 43 || code.length > 128) {
                         ^
TypeError: Cannot read properties of undefined (reading 'length')
```
A defined error message would make it easier to know what is happening

**Comment by themooks** [comment]
All feedback taken into account

**Review by carnellj-genesys — APPROVED** [review]

**Review by carnellj-genesys — APPROVED** [review]

**Review by HemanthDogiparthi12 — APPROVED** [review]

**Pull request merged**